tempesta_fw/connection.c

```diff
@@ -90,13 +91,22 @@ tfw_connection_release(TfwConnection *conn)
  * for the life of @conn instance. The socket itself may have been
  * closed, but not deleted. ss_send() makes sure that data is sent
  * only on an active socket.
- *
- * @unref_data is true if we won't use @msg any more.
  */
 int
-tfw_connection_send(TfwConnection *conn, TfwMsg *msg, bool unref_data)
+tfw_connection_send(TfwConnection *conn, TfwMsg *msg)
 {
-	return ss_send(conn->sk, &msg->skb_list, unref_data);
+	int flags = 0;
+	const TfwHttpMsg *hm = (TfwHttpMsg *)msg;
+
+	if (TFW_CONN_TYPE(conn) & Conn_Clnt) {
+		if (tfw_http_msg_conn(hm) == TFW_HTTP_CONN_CLOSE)
+			flags |= SS_SEND_F_CONN_CLOSE;
+		if (!(hm->flags & TFW_HTTP_DO_NOT_PASS))
```

Comment:
Actually, the only place where `tfw_cli_conn_send` called with indirect argument is:
https://github.com/tempesta-tech/tempesta/blob/master/tempesta_fw/http.c#L1112

But, in that case `unref_data` is the result of `bool unref_data = tfw_cache_add(resp, req);`. The latter is always `true` as `tfw_cache_add` do not returns `false` at all.